src/nom.rs

```diff
@@ -602,9 +602,10 @@ pub fn float(input: &[u8]) -> IResult<&[u8],f32> {
     recognize!(
       tuple!(
         opt!(alt!(tag!("+") | tag!("-"))),
-        alt!(
-          delimited!(digit, tag!("."), opt!(digit))
-          | delimited!(opt!(digit), tag!("."), digit)
+        alt_complete!(
```

Comment:
using `alt_complete!` can break the behaviour of the parser. If the data is`123.4` but you get it as `123` then `.4` (because of buffering, network packets or other reasons), you would parse it as `123` then a parser that fails on the unexpected `.4`.

I used it in benches/json.rs for convenience, but in the general case it does not work :/

Reply:
Should the parser differentiate between `IncompleteOrDone` or `IncompleteOrError` - which would allow the `complete!` macro to resolve it?